src/components/calcite-panel/calcite-panel.tsx

```diff
@@ -0,0 +1,202 @@
+import { Component, Element, Event, EventEmitter, Host, Prop, Watch, h } from "@stencil/core";
+import { CSS, TEXT } from "./resources";
+import { getElementDir } from "../utils/dom";
+import classnames from "classnames";
+import { CSS_UTILITY } from "../utils/resources";
+import { VNode } from "@stencil/core/dist/declarations";
+import { CalciteTheme } from "../interfaces";
+import CalciteIcon from "../utils/CalciteIcon";
+import { x16 } from "@esri/calcite-ui-icons";
+
+const SLOTS = {
+  headerContent: "header-content",
+  headerLeadingContent: "header-leading-content",
+  headerTrailingContent: "header-trailing-content",
+  footer: "footer"
+};
+
+/**
+ * @slot header-content - A slot for adding content in the center of the header.
+ * @slot header-leading-content - A slot for adding a `calcite-action` on the leading side of the header.
+ * @slot header-trailing-content - A slot for adding a `calcite-action` on the trailing side of the header.
+ * @slot footer - A slot for adding `calcite-actions` to the footer.
+ */
+@Component({
+  tag: "calcite-panel",
+  styleUrl: "calcite-panel.scss",
+  shadow: true
+})
+export class CalcitePanel {
+  // --------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  // --------------------------------------------------------------------------
+
+  /**
+   * Hides the panel.
+   */
+  @Prop({ reflect: true }) dismissed = false;
```

Comment:
Pebble: This should probably be @State() since it's mutating. You can reflect a state value to a prop by setting it on the host `<Host dismissed={this.dismissed}>`

Reply:
> This should probably be @State() since it's mutating.

@pr3tori4n Actually, I think I just need to set it as `mutable` since that is what we want. A user will need to be able to remove the dismissed prop to show it again.

Its like an input or a checkbox where a user can modify the value by interacting with the component or by modifying the property.

> You can reflect a state value to a prop by setting it on the host <Host dismissed={this.dismissed}>

Isn't that just a hacky way of making it mutable? Seems like it should just be declared mutable right?

Reply:
It's semantics. Everything is an object property in the raw JS language. State is a type of prop that's intended to track changes internal to the component, where as props are more of its options and configuration set from outside.

The React community has a lot of reference on this.
https://github.com/uberVU/react-guide/blob/master/props-vs-state.md

I ran into this when working on the the watcher for the pick-list-item selected prop. My learnings from that were that props should be treated as downward flow, set by the implementer or a higher-order component. Ideally, a component shouldn't mutate it's own props. State should hold dynamic data that changes internally. This split in responsibility helps avoid certain challenges. With the pick-list-item, watching for changes to the `selected` attribute (listening for changes from outside the component) conflicted when I was using the same property for modifying selected state from inside the component. The solution was to have a separate prop and state, and have changes to the prop update the state.

I'd say it's not necessary for internal state changes to reflect back to the DOM. But if we want that to work, it's possible.

In summary, it might work fine in some cases to have mutable props and this might be one of those cases, hence the pebble status. So you don't have to change it. But I do think it's a best practice that will help if the component grows in complexity. Your call.

Reply:
> props are more of its options and configuration set from outside.

@pr3tori4n Props aren't just set from the outside in HTMLElements. `<input>`'s `value` is an example of a mutable prop. It can be set from inside as well as outside. This isn't the same as react.

>  props should be treated as downward flow, set by the implementer or a higher-order component. Ideally, a component shouldn't mutate it's own props. 

Where are you getting this from? I couldn't find anything about this with custom elements/web components. I found a lot of the opposite where props are being mutated. Native input elements work this way. 

> I'd say it's not necessary for internal state changes to reflect back to the DOM. But if we want that to work, it's possible.

It is necessary. otherwise how would someone know whether the panel was dismissed already or not? How does someone know whether one of the `pick-list-items` is `selected` already?

> But I do think it's a best practice that will help if the component grows in complexity. Your call.

Why is it a best practice to not use a mutable prop? I think there's some confusion on whether mutable props are acceptable or not. It seems like if the updated value should be exposed then mutable is the way to go.

If I have internal state for this, I wouldn't be able to expose the updated value to the user.

@jcfranco do you have an opinion here?

Reply:
Its also worth noting `calcite-components` and `ionic` use a lot of mutable props for stateful props that need to be exposed to the user.

https://github.com/Esri/calcite-components/search?q=mutable%3A&unscoped_q=mutable%3A

https://github.com/ionic-team/ionic/search?q=mutable%3A&unscoped_q=mutable%3A

Reply:
> Where are you getting this from?

As I mentioned, my learning from the experience with the pick-list-item. If you're adding a watcher where you're reacting to a prop being modified from outside a component, then you'll want a separate state property to track internal changes. Outside of that scenario, it may not be necessary

> Why is it a best practice to not use a mutable prop?

This is my opinion, based on my understood definition of `component state` and patterns that exist in other systems. When you're mutating a prop internally, you're treating a prop as state. This violates the single responsibility principle. The prop is being used both as a means of configuration (including default value + configuration modifications) and to track internal state changes. 

Splitting these into two separate properties prevents conflicts from these two operations. One real example I ran into was with the pick-list-item. If there's a watcher on a prop reacting to external changes, it was also undesirably reacting to internal changes. This may very well be how the browser handles things internally.

But it's just a suggestion, and if you're not running into these problems because the complexity isn't there you can continue using a mutable prop if things are functional. I'm not dying on this hill - you're free to move forward with this PR either way you choose.

> <input>'s value is an example of a mutable prop

Input's value attribute and property aren't kept in sync. The value attribute is used to set the initial value of the form. The property then updates with edits from the user, but the attribute doesn't reflect those changes. This is a good example of an attribute being used for configuration but the property being treated as state and only updating internally.
https://javascript.info/dom-attributes-and-properties#property-attribute-synchronization

Reply:
> Input's value attribute and property aren't kept in sync.

@pr3tori4n I'm not referring to the `attribute` on an input, that's a different topic that would be handled by the `reflect` option.

I'm referring to the `property`.  The `value`  `property` is updated when a user interacts with the element. It's not just internal state, its a mutable property.

Reply:
> this is a good example of an attribute being used for configuration but the property being treated as state and only updating internally.

input value isn't just state. It doesn't just update internally. It can be updated internally or externally.